shared/fs/common/path-item-info.tsx

```diff
@@ -1,58 +1,122 @@
 import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
 import * as Styles from '../../styles'
 import * as Kb from '../../common-adapters'
-import {formatTimeForFS} from '../../util/timestamp'
+import LastModifiedLine from './last-modified-line-container'
+import TlfInfoLine from './tlf-info-line-container'
+import PathItemIcon from './path-item-icon-container'
+import CommaSeparatedName from './comma-separated-name'
+import * as Container from '../../util/container'
+import {useFsChildren, useFsPathMetadata, useFsOnlineStatus} from './hooks'
 
-export type PathItemInfoProps = {
-  lastModifiedTimestamp?: number
-  lastWriter?: string
-  mode: 'row' | 'default' | 'menu'
+type Props = {
+  containerStyle?: Styles.StylesCrossPlatform
+  showTooltipOnName: boolean
+  path: Types.Path
 }
 
-const Username = ({mode, lastWriter}) =>
-  mode === 'row' && Styles.isMobile ? (
-    <Kb.Text type="BodySmall">{lastWriter}</Kb.Text>
-  ) : (
-    <Kb.ConnectedUsernames
-      type="BodySmallSecondaryLink"
-      usernames={[lastWriter]}
-      inline={true}
-      onUsernameClicked="profile"
-      underline={true}
-    />
+const getChildrenNumbers = (
+  pathItems: Types.PathItems,
+  path: Types.Path
+): {folders: number; files: number} => {
+  const pathItem = pathItems.get(path, Constants.unknownPathItem)
+  return pathItem.type === Types.PathType.Folder
+    ? pathItem.children.reduce(
+        ({folders, files}, p) => {
+          const isFolder =
+            pathItems.get(Types.pathConcat(path, p), Constants.unknownPathItem).type === Types.PathType.Folder
+          return {
+            files: files + (isFolder ? 0 : 1),
```

Comment:
this isn't technically correct, is it? If the item is unknownPathItem (or anything else that isn't a file or a folder) then this is wrong